src/Package.zig

```diff
@@ -600,7 +600,7 @@ fn unpackTarball(
     var decompress = try compression.decompress(gpa, br.reader());
     defer decompress.deinit();
 
-    try std.tar.pipeToFileSystem(out_dir, decompress.reader(), .{
+    try std.tar.pipeToFileSystem(gpa, out_dir, decompress.reader(), .{
```

Comment:
I agree. But if we want to support long file names, it is necessary to have an allocator.  See @truemedian's comment https://github.com/ziglang/zig/pull/15382#discussion_r1174261182

Reply:
@travisstaloch it might be possible to use a buffer of size `std.fs.MAX_NAME_BYTES` and read the path one-component-at-a-time (erroring if any component is > `MAX_NAME_BYTES` long since it won't be possible to create it on the filesystem), using the `Dir` of the last created component to create the next component and so on.

Reply:
`MAX_NAME_BYTES` may work, but it needs to be able to reasonably handle windows systems (where such value is 260). You can't use the directory of the last component because tar files aren't necessarily laid out as a tree, you always have to create the file from the extraction root.

Reply:
260 is `MAX_PATH` on Windows (and is irrelevant to Zig since we use extended-length path APIs exclusively). See `std.os.NAME_MAX`:

https://github.com/ziglang/zig/blob/f5239677e27a25beb450d2361fb15987d56de053/lib/std/os/windows.zig#L3595-L3611

I'm not sure what you mean by the second part. If each file in a tar has a filepath from the root, then, for example:

```
some-long-component/some-long-component/some-long-component/some-long-component/some-long-component/foo.txt
```

I'm suggesting something like:

- Read into `[MAX_NAME_BYTES]u8` until either you hit a path separator or run out of room (in which case, fatal error)
- Use a temporary buffer of one byte to check if this is the last component (read into it until you hit a non-slash or end-of-filepath)
- If this is not the last component, create and open it as a directory (using either the `Dir` of the root or the last created component's `Dir`)
  + If this is the last component, create it as a file and move to the next
- If there are more components to read, put the byte from the temporary buffer into the `[MAX_NAME_BYTES]u8` buffer and repeat from the start